app/src/api/clientApi.ts

```diff
@@ -25,8 +25,51 @@ export interface SendProposalMessageRequest {
 
 export interface SendProposalMessageResponse {}
 
+export enum ProposalActionType {
+  ExternalFunctionCall = 'ExternalFunctionCall',
+  Transfer = 'Transfer',
+  SetNumApprovals = 'SetNumApprovals',
+  SetActiveProposalsLimit = 'SetActiveProposalsLimit',
+  SetContextValue = 'SetContextValue',
+  DeleteProposal = 'DeleteProposal',
+}
+
+export type FormActionType =
+  | 'Cross contract call'
+  | 'Transfer'
+  | 'Set context variable'
+  | 'Change number of approvals needed'
+  | 'Change number of maximum active proposals';
+
+export interface ExternalFunctionCallAction {
+  type: ProposalActionType.ExternalFunctionCall;
+  receiver_id: string;
+  method_name: string;
+  args: Record<string, any>;
+  deposit: string;
+  gas?: string;
+}
+
+export interface TransferAction {
+  type: ProposalActionType.Transfer;
+  amount: string;
+}
+
 export interface CreateProposalRequest {
-  receiver: String;
+  action_type: string;
+  params: {
```

Comment:
Based on proposal type request different params

Reply:
The PR is still in draft since we need to do a couple of more things to be representative and to showcase all of the features.